modules/custom/d_p_reference_content/d_p_reference_content.module

```diff
@@ -0,0 +1,75 @@
+<?php
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\paragraphs\Entity\Paragraph;
+use Drupal\field\Entity\FieldConfig;
+use \Drupal\Core\Template\Attribute;
+
+/**
+ * Implements hook_entity_presave().
+ */
+function d_p_reference_content_entity_presave(EntityInterface $entity) {
+  $type = $entity->bundle();
+  switch ($type) {
+    case 'd_p_reference_content':
+      if ($entity instanceof Paragraph) {
+        $reference_content = $entity->get('field_d_p_reference_content');
+        $values = $reference_content->getValue();
+
+        /** @var FieldConfig $def */
+        $def = $reference_content->getDataDefinition();
+        $cardinality = $def->getFieldStorageDefinition()->getCardinality();
+
+        // Get latest blog node nids.
+        // Selected nodes are excluded from results.
+        $auto_values = \Drupal::service('d_p_reference_content.get_content')
+          ->getSortedContentByType('blog_post', 'created', 'DESC', $values);
+        $merged_values = array_merge($values, $auto_values);
```

Comment:
Tu nie będzie tak do końca zgodnie z kryteriami akceptacji - bo jeśli:

* Użytkownik wypełni np. 1 pole z referencją i zostawi pozostałe puste
* Użytkownik zapisze paragraf
* Użytkownik doda 5 nowych artykułów w blogu

To paragraf pokaże jeden wybrany artykuł użytkownika i trzy dodatkowe, które były nowe w momencie zapisania paragrafu, ale teraz już są stare. Powinniśmy dołączać te nowe artykuły na etapie pokazywania paragrafu.